braindecode/models/eegconformer.py

```diff
@@ -0,0 +1,272 @@
+
+    .. versionadded:: 0.8
+
+    We aggregate the parameters based on the parts of the models.
+
+    Parameters PatchEmbedding
+    -------------------------
+    - n_kernel: int
+        Number of kernels for the temporal convolution layer (first layer).
+    - n_kernel_temp_conv: int
+        Kernel size of the temporal convolution layer.
+    - n_kernel_spatial_conv: int
+        kernel size of the spatial convolution layer.
+    - n_kernel_avg_pool: int
+        Kernel size of the average pooling layer.
+    - stride_avg_pool: int
+        Stride of the average pooling layer.
+    - drop_prob: float
+        Dropout rate of the convolutional layer.
+
+    Parameters TransformerEncoder
+    -----------------------------
+    - att_depth: int
+        Number of self-attention layers.
+    - att_heads: int
+        Number of attention heads.
+    - att_drop_prob: float
+        Dropout rate of the self-attention layer.
+
+    Parameters ClassificationHead
+    -----------------------------
+    - fc_dim: int
+        The dimension of the fully connected layer.
+    - n_classes: int
+        Number of classes.
+
+
+    References
+    ----------
+    .. [EEG Conformer] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       IEEE Transactions on Neural Systems and Rehabilitation Engineering,
+       31, pp.710-719. https://ieeexplore.ieee.org/document/9991178
+    .. [EEG Conformer Code] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       https://github.com/eeyhsong/EEG-Conformer.
+
+
+    """
+
+    def __init__(
+            self,
+            n_kernel=40,
+            n_kernel_temp_conv=25,
+            n_kernel_spatial_conv=22,
+            n_kernel_avg_pool=75,
+            stride_avg_pool=15,
+            drop_prob=0.5,
+            att_depth=6,
+            att_heads=10,
+            att_drop_prob=0.5,
+            fc_dim=2440,
+            n_classes=4,
+    ):
+        super().__init__(
+            PatchEmbedding(
+                n_kernel,
+                n_kernel_temp_conv,
+                n_kernel_spatial_conv,
+                n_kernel_avg_pool,
+                stride_avg_pool,
+                drop_prob,
+            ),
+            TransformerEncoder(att_depth, n_kernel, att_heads, att_drop_prob),
+            ClassificationHead(n_kernel, fc_dim, n_classes),
+        )
+
+
+class PatchEmbedding(nn.Module):
+    """Patch Embedding.
+
+    The authors used a convolution moduleto capture local features,
+    instead of postion embedding.
+
+    """
+
+    def __init__(
+            self,
+            n_kernel,
+            n_kernel_temp_conv,
+            n_kernel_spatial_conv,
+            n_kernel_avg_pool,
+            stride_avg_pool,
+            drop_prob,
+    ):
+        super().__init__()
+
+        self.shallownet = nn.Sequential(
```

Comment:
How about using `ShallowFBCSPNet` already implemented in the package? To remove `conv_classifier` and other not needed layers you may use sth like that:
```python
model = ShallowFBCSPNet(in_chans=22, 40, conv_nonlin=nn.ELU(), first_conv_bias=True)
new_model = nn.Sequential()
for name, module_ in model.named_children():
    if "conv_classifier" in name:
        break
    new_model.add_module(name, module_)
model = new_model
```

Reply:
What do you think @eeyhsong? Make sense for you?

Reply:
I was thinking more about this, and for me, it is best if we keep separate in another PR for two reason:
- 1) Given the tokenization process, the number of channels is always "equals" 1.  So a different network input as the traditional input in the ShallowNet. When compared with the original shallow, it potentially can lead to a different network, as already pointed out in the case of the attention layer and/or the addition of the transformer layer.
- 2) ShallowNet will be refactored soon in code sprint by @robintibor to remove the 4d entry used in the cropped option.

All these points can be optimized later with a better standardizing of the variables and modules.